examples/tensorflow/token_classification/run_ner.py

```diff
@@ -0,0 +1,477 @@
+        if model_args.model_name_or_path:
+            model = TFAutoModelForTokenClassification.from_pretrained(
+                model_args.model_name_or_path,
+                config=config,
+            )
+        else:
+            logger.info("Training new model from scratch")
+            model = TFAutoModelForTokenClassification.from_config(config)
+
+        model.resize_token_embeddings(len(tokenizer))
+        # endregion
+
+        # region Create TF datasets
+        num_replicas = training_args.strategy.num_replicas_in_sync
+        total_train_batch_size = training_args.per_device_train_batch_size * num_replicas
+        train_batches_per_epoch = len(train_dataset) // total_train_batch_size
+        tf_train_dataset = dataset_to_tf(
+            train_dataset,
+            tokenizer,
+            total_batch_size=total_train_batch_size,
+            num_epochs=training_args.num_train_epochs,
+            shuffle=True,
+        )
+        total_eval_batch_size = training_args.per_device_eval_batch_size * num_replicas
+        eval_batches_per_epoch = len(eval_dataset) // total_eval_batch_size
+        tf_eval_dataset = dataset_to_tf(
+            eval_dataset,
+            tokenizer,
+            total_batch_size=total_eval_batch_size,
+            num_epochs=training_args.num_train_epochs,
+            shuffle=False,
+        )
+
+        # endregion
+
+        # region Optimizer, loss and compilation
+        optimizer, lr_schedule = create_optimizer(
+            init_lr=training_args.learning_rate,
+            num_train_steps=int(training_args.num_train_epochs * train_batches_per_epoch),
+            num_warmup_steps=training_args.warmup_steps,
+            adam_beta1=training_args.adam_beta1,
+            adam_beta2=training_args.adam_beta2,
+            adam_epsilon=training_args.adam_epsilon,
+            weight_decay_rate=training_args.weight_decay,
+        )
+
+        def dummy_loss(y_true, y_pred):
+            return tf.reduce_mean(y_pred)
+
+        model.compile(loss={"loss": dummy_loss}, optimizer=optimizer)
+        # endregion
+
+        # Train!
+        logger.info("***** Running training *****")
+        logger.info(f"  Num examples = {len(train_dataset)}")
+        logger.info(f"  Num Epochs = {training_args.num_train_epochs}")
+        logger.info(f"  Instantaneous batch size per device = {training_args.per_device_train_batch_size}")
+        logger.info(f"  Total train batch size = {total_train_batch_size}")
+        # Only show the progress bar once on each machine.
+        model.fit(
+            tf_train_dataset,
+            validation_data=tf_eval_dataset,
+            epochs=int(training_args.num_train_epochs),
+            steps_per_epoch=train_batches_per_epoch,
+            validation_steps=eval_batches_per_epoch,
+        )
+
+    # We don't do predictions in the strategy scope because there are some issues in there right now.
+    # They'll get fixed eventually, promise!
+
```

Comment:
Can we add a quick and dirty evaluation that gets the result of predict and passes it to `metric.compute()`? A bit like the `compute_metrics` function in the PyTorch example.

Reply:
I can probably do that quickly tomorrow, sure!